packages/synapse-react-client/src/components/TeamSubjectsSelector/TeamSubjectsSelector.test.tsx

```diff
@@ -0,0 +1,199 @@
+    renderComponent({
+      ...defaultProps,
+      subjects: [],
+    })
+    expect(screen.getByText(NO_TEAMS_SELECTED)).toBeVisible()
+  })
+
+  test('removes a team and drops any entity subjects', async () => {
+    const { user } = await setUp()
+
+    const removeButton = screen.getByRole('button', { name: REMOVE_TEXT })
+    await user.click(removeButton)
+
+    // entity subject is not included
+    expect(onUpdate).toHaveBeenCalledTimes(1)
+    expect(onUpdate).toHaveBeenCalledWith([])
+  })
+
+  test('adds teams and drops any entity subjects', async () => {
+    const { user } = await setUp()
+
+    const newTeamIds = `${MOCK_TEAM_ID_2}, ${MOCK_TEAM_ID_3}`
+    await addTeams(newTeamIds, user)
+
+    expect(onUpdate).toHaveBeenCalledTimes(1)
+    expect(onUpdate).toHaveBeenCalledWith([
+      teamSubject,
+      { id: MOCK_TEAM_ID_2.toString(), type: RestrictableObjectType.TEAM },
+      { id: MOCK_TEAM_ID_3.toString(), type: RestrictableObjectType.TEAM },
+    ])
+
+    const input = screen.getByPlaceholderText(HELP_TEXT)
+    expect(input).toHaveValue('')
+    expect(onUpdateTeamIDsTextbox).toHaveBeenLastCalledWith('')
+  })
+
+  test('drops duplicates when adding new teams', async () => {
+    const { user } = await setUp()
+
+    // add 2 new IDs, 1 duplicate of a new ID
+    const newTeamIds = `${MOCK_TEAM_ID_2}, ${MOCK_TEAM_ID_3}, ${MOCK_TEAM_ID_2}`
+    await addTeams(newTeamIds, user)
+
+    expect(onUpdate).toHaveBeenCalledTimes(1)
+    expect(onUpdate).toHaveBeenCalledWith([
+      teamSubject,
+      { id: MOCK_TEAM_ID_2.toString(), type: RestrictableObjectType.TEAM },
+      { id: MOCK_TEAM_ID_3.toString(), type: RestrictableObjectType.TEAM },
+    ])
+
+    const input = screen.getByPlaceholderText(HELP_TEXT)
+    expect(input).toHaveValue('')
+    expect(onUpdateTeamIDsTextbox).toHaveBeenLastCalledWith('')
+  })
+
+  test('displays an error when attempting to add a team that was previously added', async () => {
+    const { user } = await setUp()
+
+    const newTeamIds = `${MOCK_TEAM_ID_2}, ${MOCK_TEAM_ID}`
+    await addTeams(newTeamIds, user)
+
+    const alert = await screen.findByRole('alert')
+    expect(alert).toHaveTextContent(
+      TEAM_ALREADY_SELECTED(MOCK_TEAM_ID.toString()),
+    )
+
+    expect(onUpdate).not.toHaveBeenCalled()
+    expect(onUpdateTeamIDsTextbox).toHaveBeenLastCalledWith(newTeamIds)
+  })
+
+  test('displays an error when error parsing team ids', async () => {
+    const { user } = await setUp()
+
+    const invalidTeamId = 'abc'
+    const newTeamIds = `${MOCK_TEAM_ID_2}, ${invalidTeamId}`
+    await addTeams(newTeamIds, user)
+
+    const alert = await screen.findByRole('alert')
+    expect(alert).toHaveTextContent(TEAM_PARSING_ERROR(invalidTeamId))
+
+    expect(onUpdate).not.toHaveBeenCalled()
+    expect(onUpdateTeamIDsTextbox).toHaveBeenLastCalledWith(newTeamIds)
+  })
+
+  test('displays an error when team is not found', async () => {
```

Comment:
Currently, SWC allows the user to add teams to an AR that are not found. In SWC, teams that aren't found are shown as errors -- 
<img width="833" alt="SWC-6730_teamSubjectsSelector_teamNotFound_SWC" src="https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/26949006/d2b7b52c-1d9a-43dc-8c97-0d8c7b31446f">

Implement similar behavior here where an error badge is displayed instead of the standard TeamBadge when a team is not found -- 

https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/26949006/7beb8d4b-b699-49bd-b635-ea0912e3b765